### DISCUSSION TITLE
DSL vs Workflow-as-code (Wac) - which is right?

### DISCUSSION THREAD

**Discussion by globalflea** [serverlessworkflow/specification#541] [General]
in another discussion with tsurdilo, he shared a youtube video when the serverless spec was implemented on temporal.io (https://github.com/serverlessworkflow/specification/discussions/526)

I watched the video and was intrigued by temporal.io, as it provides a lot of the fundamental features required for long-running workflows.  Digging a little deeper, however, i found that temporal introduces the concept of "Workflow as code" (as with a lot of other "... as code" recently).  This seems to be a big contrast to the DSL approach and i am now wondering if the DSL approach still make sense - i.e. which is better DSL or Workflow-as-code (WaC)??

**Comment by tsurdilo** [comment]
Yes, in Temporal you write workflows as code, but you can run any DSL on top of Temporal. There is a Java sample that runs our DSL on top of Temporal [here](https://github.com/temporalio/samples-java/tree/main/src/main/java/io/temporal/samples/dsl). We are also working on out-of-box integration with Java SDK (repo still private atm but will be public soon) and if there is adoption will add it for TypeScript and Go SDKs as well in the future. 

As far as how you should write your workflows, thats really up to you. [Here](https://docs.temporal.io/blog/defining-workflows) is a blog I wrote that kinda talks about pros/cons of major used ways of writing workflows.

**Reply by thiagoanselmo** [reply]
SDKs TypeScript <3

**Comment by tsurdilo** [comment]
@globalflea let me know if you want to be added to our currently private repo for Temporal integration (Java). No need to contribute, but maybe you can get some ideas about a possible Go impl, and I can add you to slack channel where the discussions are happening as well. Also feel free to ping me or @antmendoza with questions about integration.

**Reply by globalflea** [reply]
Yes, think I would appreciate it. Thanks

**Reply by namp10010** [reply]
hi @tsurdilo and @globalflea,

I have been looking for a Go implementation similar to the [Java one](https://github.com/temporalio/samples-java/tree/main/src/main/java/io/temporal/samples/dsl) to run Serverless workflow on top of Temporal. Just wonder if there is any any example that you can point me to?

**Reply by aaronchipper** [reply]
I would also be very interested in working on a Go implementation.
@namp10010 - perhaps we could collaborate on this?

**Reply by namp10010** [reply]
@aaronchipper , I am keen on exploring this if there's no existing one. I think, like the java example, we can start with a small subset of Serverless Workflow spec.

**Reply by murat-shiftright** [reply]
Hi @tsurdilo, I don't think this private repo has become public yet.   I am interested in Java integration and have looked into it to some extent.  Could you please add me to the private repo?

**Reply by giorgosera** [reply]
I am also interested in this but for Typescript. I can contribute if one doesn't exist or needs work.

**Reply by antmendoza** [reply]
@giorgosera it does not exist for typescript, but if you are interested we can work on it together

**Reply by giorgosera** [reply]
@antmendoza for sure! How can we get access to the Java project so we can understand how this might work for Typescript too?

**Reply by tsurdilo** [reply]
You can look at https://github.com/temporalio/samples-java/tree/main/src/main/java/io/temporal/samples/dsl

We are not yet ready to make the repo you reference public but it takes the same approach. hope this helps

**Reply by giorgosera** [reply]
Yes, we've already started working based on that example. Thanks @tsurdilo!

**Reply by thiagoanselmo** [reply]
I can contribute for Typescript

**Reply by thiagoanselmo** [reply]
@giorgosera 

Where is the repo? I can contribute to this!

**Reply by tsurdilo** [reply]
@thiagoanselmo hey maybe idea could be to add this ts integration in your repo and share it, then we could work on moving it to a SW repo if you want to contribute it. wdyt?

**Reply by thiagoanselmo** [reply]
@tsurdilo 

Hi, yes I can!
How did you start the SDK in JAVA.
Send me the links you used that helped you build the SDK.

Or if you already have something in your mind where to start, please direct me!

**Reply by antmendoza** [reply]
@thiagoanselmo  if you are still interested in typescript<->Temporal we can work on this together.

**Reply by vsai-zuora** [reply]
@tsurdilo I can help with temporal integration (java). I am currently working on building an out-of-the-box DSL-based implementation on top of temporal using SW at my company. Hope I can contribute with some of the ideas.

**Comment by tsurdilo** [comment]
I'd really like all of your help with the Temporal integration work. Can you please give indication on how much time per week you would be able to work on this (just ballpark is enough) and what SDK you want to help with and let's start planning for this.

**Reply by SinanLai** [reply]
Hi, @tsurdilo, our backend project now uses the Temporal workflow but we want to introduce Serverless to improve the usage of hardware. I am very interested in this and want to know whether the Temporal integration work started. If not, is there any plan for this? thx

**Comment by thiagoanselmo** [comment]
Guys, I'm starting soon!
I'm finishing up a log centering project.

Since I didn't want to use graylog because it uses Java (I think it's heavy) and uses elasticsearch for searches, I think it's expensive to maintain an elastic cluster.

So in this project he uses Atlas Search (lucene).
I finish this project this week, and we're going to make it open source as well.

And next week I'm going to start this project!

Sorry I'm late getting started, but I had other demands I needed to fulfill!
